code/datums/status_effects/xeno_buffs.dm

```diff
@@ -24,6 +27,302 @@
 	owner.heal_limb_damage(0, 5)
 	return ..()
 
+// ***************************************
+// *********** Essence Link
+// ***************************************
+/obj/effect/ebeam/essence_link
+	name = "essence link beam"
+
+/datum/status_effect/stacking/essence_link
+	id = "xeno_essence_link"
+	tick_interval = 2 SECONDS
+	stacks = 0
+	stack_decay = -1 //Not meant to decay.
+	max_stacks = 3
+	consumed_on_threshold = FALSE
+	consumed_on_fadeout = FALSE
+	alert_type = null
+	/// The owner of the link.
+	var/mob/living/carbon/xenomorph/link_owner
+	/// Whom the owner is linked to.
+	var/mob/living/carbon/xenomorph/link_target
+	/// References the Essence Link action and its vars.
+	var/datum/action/xeno_action/activable/essence_link/essence_link_action
+	/// If the target xeno was within range.
+	var/was_within_range = TRUE
+	/// Cooldown for passive attunement increase.
+	COOLDOWN_DECLARE(attunement_increase)
+	/// Delay between plasma warnings. Plasma warnings are issued if there's not enough plasma for the passive regeneration.
+	var/plasma_warning_cooldown = 20 SECONDS
+	/// Cooldown for the plasma warning.
+	COOLDOWN_DECLARE(plasma_warning)
+	/// The beam used to represent the link between linked xenos.
+	var/datum/beam/current_beam
+
+/datum/status_effect/stacking/essence_link/on_creation(mob/living/new_owner, stacks_to_apply, mob/living/carbon/link_target)
+	link_owner = new_owner
+	src.link_target = link_target
+	essence_link_action = link_owner.actions_by_path[/datum/action/xeno_action/activable/essence_link]
+	ADD_TRAIT(link_owner, TRAIT_ESSENCE_LINKED, TRAIT_STATUS_EFFECT(id))
+	ADD_TRAIT(link_target, TRAIT_ESSENCE_LINKED, TRAIT_STATUS_EFFECT(id))
+	RegisterSignal(link_owner, COMSIG_MOB_DEATH, .proc/handle_death)
+	RegisterSignal(link_target, COMSIG_MOB_DEATH, .proc/handle_death)
+	toggle_link(TRUE)
+	to_chat(link_target, "[link_owner] has established an Essence Link with you. Stay within [DRONE_ESSENCE_LINK_RANGE] tiles to maintain it.")
+	return ..()
+
+/datum/status_effect/stacking/essence_link/add_stacks(stacks_added)
+	. = ..()
+	essence_link_action.update_button_icon()
+	link_owner.balloon_alert(link_owner, "Attunement: [stacks]/[max_stacks]")
+	COOLDOWN_START(src, attunement_increase, essence_link_action.attunement_cooldown)
+	update_beam()
+
+/datum/status_effect/stacking/essence_link/on_remove()
+	. = ..()
+	link_owner.balloon_alert(link_owner, "Essence Link cancelled")
+	link_target.balloon_alert(link_target, "Essence Link cancelled")
+	toggle_link(FALSE)
+	essence_link_action.end_ability()
+	REMOVE_TRAIT(link_owner, TRAIT_ESSENCE_LINKED, TRAIT_STATUS_EFFECT(id))
+	REMOVE_TRAIT(link_target, TRAIT_ESSENCE_LINKED, TRAIT_STATUS_EFFECT(id))
+	UnregisterSignal(link_owner, list(COMSIG_MOB_DEATH, COMSIG_XENOMORPH_RESIN_JELLY_APPLIED, COMSIG_XENOMORPH_BRUTE_DAMAGE, COMSIG_XENOMORPH_BURN_DAMAGE))
```

Comment:
toggling link off already cuts jelly and damage

Reply:
Whoops.